homeassistant/components/smartthings/fan.py

```diff
@@ -92,6 +142,33 @@ def is_on(self) -> bool:
         return self._device.status.switch
 
     @property
-    def percentage(self) -> int:
+    def percentage(self) -> int | None:
         """Return the current speed percentage."""
-        return ranged_value_to_percentage(SPEED_RANGE, self._device.status.fan_speed)
+        if FanEntityFeature.SET_SPEED in self._attr_supported_features:
+            return ranged_value_to_percentage(
+                SPEED_RANGE, self._device.status.fan_speed
+            )
+
+        return None
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., auto, smart, interval, favorite.
+
+        Requires FanEntityFeature.SET_SPEED.
+        """
+        if FanEntityFeature.PRESET_MODE in self._attr_supported_features:
```

Comment:
As above. We can just return the device status or possibly guard for not able to return it?